tests/backends/data/test_async_mongo.py

```diff
@@ -0,0 +1,1087 @@
+    msg = (
+        f"The target=`foo{invalid_character}bar` is not a valid database name: "
+        f"database names cannot contain the character '{invalid_character}'"
+    )
+
+    with pytest.raises(BackendParameterException, match=msg):
+        with caplog.at_level(logging.ERROR):
+            async for result in backend.list(f"foo{invalid_character}bar"):
+                next(result)
+
+    assert (
+        "ralph.backends.data.async_mongo",
+        logging.ERROR,
+        msg,
+    ) in caplog.record_tuples
+
+
+@pytest.mark.anyio
+async def test_backends_data_async_mongo_data_backend_list_method_with_failure(
+    async_mongo_backend, monkeypatch, caplog
+):
+    """Test the `AsyncMongoDataBackend.list` method given a failure while retrieving
+    MongoDB collections, should raise a `BackendException`.
+    """
+
+    def mock_list_collections():
+        """Mock the `list_collections` method always raising an exception."""
+        raise PyMongoError("Connection error")
+
+    backend = async_mongo_backend()
+    monkeypatch.setattr(backend.database, "list_collections", mock_list_collections)
+    msg = "Failed to list MongoDB collections: Connection error"
+    with pytest.raises(BackendException, match=msg):
+        with caplog.at_level(logging.ERROR):
+            async for result in backend.list():
+                next(result)
+
+    assert (
+        "ralph.backends.data.async_mongo",
+        logging.ERROR,
+        msg,
+    ) in caplog.record_tuples
+
+
+@pytest.mark.anyio
+async def test_backends_data_async_mongo_data_backend_list_method_without_history(
+    mongo, async_mongo_backend, monkeypatch
+):
+    """Test the `MongoDataBackend.list` method without history."""
+    # pylint: disable=unused-argument
+
+    backend = async_mongo_backend()
+
+    # Test `list` method with default parameters
+    result = [collection async for collection in backend.list()]
+    assert result == [MONGO_TEST_COLLECTION]
+
+    # Test `list` method with a given target (database for MongoDB)
+    result = [
+        collection async for collection in backend.list(target=MONGO_TEST_DATABASE)
+    ]
+    assert result == [MONGO_TEST_COLLECTION]
+
+    # Test `list` method with detailed information about collections
+    result = [collection async for collection in backend.list(details=True)]
+    assert result[0]["name"] == MONGO_TEST_COLLECTION
+
+    # Test `list` method with several collections
+    await backend.database.create_collection("bar")
+    await backend.database.create_collection("baz")
+
+    result = [collection async for collection in backend.list()]
+    assert sorted(result) == sorted([MONGO_TEST_COLLECTION, "bar", "baz"])
+
+    result = [collection["name"] async for collection in backend.list(details=True)]
+    assert sorted(result) == (sorted([MONGO_TEST_COLLECTION, "bar", "baz"]))
+
+    result = [collection async for collection in backend.list("non_existent_database")]
+    assert not list(result)
```

Comment:
```suggestion
    assert not result
```
Isn't it already a list?